cli/cmd/root.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright © 2020 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"perper/cmd/agent"
+	"perper/cmd/fabric"
+
+	"github.com/spf13/cobra"
+
+	"github.com/spf13/viper"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "perper",
+	Short: "CLI interaction with perper",
+	// Long:  `A longer description ...`,
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.perper.yaml)")
+
+	rootCmd.AddCommand(fabric.FabricCmd)
+	rootCmd.AddCommand(agent.AgentCmd)
+}
+
+func initConfig() {
+	if cfgFile != "" {
+		viper.SetConfigFile(cfgFile)
+	} else {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Search config in home directory with name ".perper" (without extension).
+		viper.AddConfigPath(home)
+		viper.SetConfigName(".perper")
+		viper.SetEnvPrefix("toml")
+	}
+
```

Comment:
```suggestion
	}
	viper.SetEnvPrefix("perper")
```

Unless I'm misunderstanding something, `SetEnvPrefix("toml")` would make it so that the config variable `foo` is matched by the environment variable `TOML_FOO`, while we would like it to be `PERPER_FOO`.